sklearn/kernel_approximation.py

```diff
@@ -590,6 +591,11 @@ class AdditiveChi2Sampler(TransformerMixin, BaseEstimator):
     0.9499...
     """
 
+    _parameter_constraints = {
+        "sample_steps": [Interval(int, 1, None, closed="left")],
+        "sample_interval": [None, Interval(float, 0, None, closed="left")],
```

Comment:
Rather than using `int` and `float`, we use `Integral` and `Real`, respectively. These can be imported from `numbers`:

```python
from numbers import Integral, Real
```